### ISSUE TITLE
Test tests/test_pyav.py::test_rotation_flag_metadata fails with a KeyError

### ISSUE THREAD

**Issue by anjos** [imageio/imageio#912]
The test `tests/test_pyav.py::test_rotation_flag_metadata` fails with a KeyError on conda-forge (reproducible locally): https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=608588&view=logs&jobId=656edd35-690f-5c53-9ba3-09c10d0bea97&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d

The issue:

```text
pytest -sv tests/test_pyav.py::test_rotation_flag_metadata
================================== test session starts ===================================
platform darwin -- Python 3.9.13, pytest-7.1.3, pluggy-1.0.0 -- /Users/andre/mamba/envs/imageio-dev/bin/python
cachedir: .pytest_cache
rootdir: /Users/andre/work/anjos/imageio-feedstock/imageio
collected 1 item

tests/test_pyav.py::test_rotation_flag_metadata Already up to date.
FAILED

======================================== FAILURES ========================================
______________________________ test_rotation_flag_metadata _______________________________

test_images = PosixPath('/Users/andre/work/anjos/imageio-feedstock/imageio/.test_images')
tmp_path = PosixPath('/private/var/folders/y4/n6m6rkls6zs72x8jd2w2tlx40000gn/T/pytest-of-andre/pytest-2/test_rotation_flag_metadata0')

    def test_rotation_flag_metadata(test_images, tmp_path):
        with iio.imopen(tmp_path / "test.mp4", "w", plugin="pyav") as file:
            file.init_video_stream("libx264")
            file.container_metadata["comment"] = "This video has a rotation flag."
            file.video_stream_metadata["rotate"] = "90"

            for _ in range(5):
                for frame in iio.imiter(test_images / "newtonscradle.gif"):
                    file.write_frame(frame)

        meta = iio.immeta(tmp_path / "test.mp4", plugin="pyav")
        assert meta["comment"] == "This video has a rotation flag."
>       assert meta["rotate"] == "90"
E       KeyError: 'rotate'

tests/test_pyav.py:454: KeyError
================================ short test summary info =================================
FAILED tests/test_pyav.py::test_rotation_flag_metadata - KeyError: 'rotate'
```

Reproducing:

```sh
$ git clone <this-repo> && cd imageio
$ mamba create -n imageio-dev python=3.9 pillow numpy pip pytest psutil tifffile gdal astropy freeimage fsspec requests av imageio-ffmpeg
$ conda activate imageio-dev
(imageio-dev) $ pip install -e .
$ pytest -sv tests/test_pyav.py::test_rotation_flag_metadata
...
        meta = iio.immeta(tmp_path / "test.mp4", plugin="pyav")
        assert meta["comment"] == "This video has a rotation flag."
>       assert meta["rotate"] == "90"
E       KeyError: 'rotate'

tests/test_pyav.py:454: KeyError
```

Any ideas?

C.f. C.f.: https://github.com/conda-forge/imageio-feedstock/pull/56

**Comment by FirefoxMetzger** [comment]
@anjos Yes, this is caused by pyav 10.0 and I've created an issue for it there (https://github.com/PyAV-Org/PyAV/issues/1045). I'm not sure what causes the problem there, but for the time being the solution here is to skip v10.0.

I updated this for our optional dependency (https://github.com/imageio/imageio/pull/894) and that part is already released, so if you install via pip you can do `pip install imageio[pyav]` and circumvent this issue. Not sure if we can do the same on conda, but if there are changes needed here, feel free to let me know and I will be happy to update :)

**Comment by anjos** [comment]
Right - I probably missed the `["av!=10.0.0"]` on the conda-forge recipe.  Let me check that.

**Comment by anjos** [comment]
@FirefoxMetzger: that fixed this error and now the conda-forge package is better synced with your upstream.  Thanks!

I'd tend to leave this issue opened until the issue is permanently gone, if that is OK with you.

**Comment by FirefoxMetzger** [comment]
I don't think there is much we can do within ImageIO to improve this, so I am not sure what we gain by keeping this issue open.   That said, I might be missing something though so if you have a specific reason in mind please do let me know :)

From my perspective, the issue is already tracked by https://github.com/PyAV-Org/PyAV/issues/1045 which is in the repo where it will have to be fixed, making this issue a duplicate. Once that is done, ImageIO and its conda package will naturally pick up the improvement and things will work as expected again without us doing any changes.

**Comment by anjos** [comment]
OK.
